src/commands/info/releasenotes/display.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// Needed this to ensure the "helpers" were decalred before read in examples
+/* eslint-disable @typescript-eslint/member-ordering */
+
+import { Env } from '@salesforce/kit';
+import { flags, SfdxCommand } from '@salesforce/command';
+import { getString } from '@salesforce/ts-types';
+import { Messages } from '@salesforce/core';
+
+import { getInfoConfig, InfoConfig } from '../../../shared/get-info-config';
+import { getReleaseNotes } from '../../../shared/get-release-notes';
+import { getDistTagVersion } from '../../../shared/get-dist-tag-version';
+
+// Initialize Messages with the current plugin directory
+Messages.importMessagesDirectory(__dirname);
+
+// Load the specific messages for this file. Messages from @salesforce/command, @salesforce/core,
+// or any library that is sfdxusing the messages framework can also be loaded this way.
+const messages = Messages.loadMessages('@salesforce/plugin-info', 'display');
+
+export default class Display extends SfdxCommand {
+  private static helpers = ['stable', 'stable-rc', 'latest', 'latest-rc', 'rc'];
+
+  public static description = messages.getMessage('commandDescription');
+
+  public static aliases = ['whatsnew'];
+
+  public static examples = [
+    `$ sfdx info:releasenotes:display
+  display release notes for currently installed CLI version
+  `,
+    `$ sfdx info:releasenotes:display --version "1.2.3"
+  display release notes for CLI version 1.2.3
+  `,
+    `$ sfdx info:releasenotes:display --version "stable-rc"
+  can be called with tag "helpers", available options are: ${Display.helpers.join(', ')}
+`,
+  ];
+
+  protected static flagsConfig = {
+    version: flags.string({
+      char: 'v',
+      description: messages.getMessage('versionFlagDescription'),
+    }),
+  };
+
+  public async run(): Promise<void> {
+    if (new Env().getBoolean('PLUGIN_INFO_HIDE_RELEASE_NOTES')) {
+      // We don't ever want to exit the process for info:releasenotes:display
```

Comment:
is that the explanation for why everything is in the try/catch/warn?

What would be the harm in throwing?  Is it because the command could be run inside the update hook/script?

Reply:
Yep exactly. Once released, this is going to run after every install/update as part of the post update hook/scripts. We don't want any of our customers' CI builds to fail because of this, just want to exit fairly silently. If they have issues beyond that, they can suppress it completely with the `PLUGIN_INFO_HIDE_RELEASE_NOTES` env var

Reply:
suggestion

1. hook runs logic (and catches everything to avoid what you're describing, like your command is doing)
2. command run logic and throws errors

As a hook, I probably have a valid version and can find release notes.  Users manually running the command could put in all sorts of garbage input.

That also saves the hook from having to do various costly tasks ("cli time").